allennlp/data/semparse/worlds/world.py

```diff
@@ -82,6 +83,52 @@ def get_valid_actions(self) -> Dict[str, List[str]]:
                                        valid_starting_types=self.get_valid_starting_types(),
                                        num_nested_lambdas=self._num_nested_lambdas)
 
+    def get_paths_to_root(self,
+                          action: str,
+                          max_path_length: int = 20,
+                          beam_size: int = 30,
+                          max_num_paths: int = 10) -> List[List[str]]:
+        """
+        For a given action, returns at most ``max_num_paths`` paths to the root (production with
+        ``@START@``) that are not longer than ``max_path_length``.
+        """
+        all_actions = self.all_possible_actions()
+        rhs_indexed_actions: Dict[str, List[Tuple[str, str]]] = defaultdict(list)
+        for possible_action in all_actions:
+            lhs, rhs = possible_action.split(' -> ')
```

Comment:
`left_side` and `right_side` instead of `lhs` and `rhs`.

Reply:
Done.